pkg/engine/task/task.go

```diff
@@ -0,0 +1,69 @@
+package task
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/apis/kudo/v1alpha1"
+)
+
+// Tasker is an interface that represents any runnable task for an operator. This method is treated
+// as idempotent and will be called multiple times during the life-cycle of the plan execution.
+// Method returns a boolean, signalizing that the task has finished successfully, and an error.
+// An error can wrap the ErrFatalExecution for errors that can not be retried e.g. failed template
+// rendering. This will result in a v1alpha1.ExecutionFatalError in the plan execution status. A normal
+// error e.g. failure during accessing the API server will be treated  as "transient", meaning plan
+// execution engine  can retry it next time. Only a (true, nil) return value will be treated as successful
+// task execution.
+type Tasker interface {
+	Run(ctx Context) (bool, error)
+}
+
+// Available engine tasks kinds
```

Comment:
the word engine is kind of extra to me there :) I would just call them task kinds :)